source/common/buffer/buffer_impl.cc

```diff
@@ -189,7 +196,7 @@ SliceDataPtr OwnedImpl::extractMutableFrontSlice() {
   slices_.pop_front();
   if (!slice.isMutable()) {
     // Create a mutable copy of the immutable slice data.
-    Slice mutable_slice{size};
+    Slice mutable_slice{size, account()};
```

Comment:
See line 207 below.  We explicitly avoid keeping drain trackers around when extracting mutable slices.  I would suggest consistency with drain trackers:
- Do not associate the slice created here with the account.
- Actually, in the else case clear the accounts associated with the slice mutable slice that is being extracted.